types/signed_message.go

```diff
@@ -28,6 +29,28 @@ func init() {
 type SignedMessage struct {
 	MeteredMessage `json:"meteredMessage"`
 	Signature      Signature `json:"signature"`
+	// Pay attention to Equals() if updating this struct.
+}
+
+// NewSignedMessage accepts a message `msg` and a signer `s`. NewSignedMessage returns a `SignedMessage` containing
+// a signature derived from the serialized `msg` and `msg.From`
+func NewSignedMessage(msg Message, s Signer, gasPrice AttoFIL, gasLimit GasUnits) (*SignedMessage, error) {
+	meteredMsg := NewMeteredMessage(msg, gasPrice, gasLimit)
+
+	bmsg, err := meteredMsg.Marshal()
```

Comment:
Total nit && not your bug: `bmsg` seems like an out of date name unless I'm missing something.